src/libraries/System.Private.CoreLib/src/System/Buffers/Binary/BinaryPrimitives.ReverseEndianness.cs

```diff
@@ -0,0 +1,260 @@
+        {
+            static abstract T Reverse(T value);
+            static abstract Vector128<T> Reverse(Vector128<T> vector);
+            static abstract Vector256<T> Reverse(Vector256<T> vector);
+        }
+
+        /// <summary>Copies every primitive value from <paramref name="source"/> to <paramref name="destination"/>, reversing each primitive by performing an endianness swap as part of writing each.</summary>
+        /// <param name="source">The source span to copy.</param>
+        /// <param name="destination">The destination to which the source elements should be copied.</param>
+        /// <remarks>The source and destination spans may overlap. The same span may be passed as both the source and the destination in order to reverse each element's endianness in place.</remarks>
+        /// <exception cref="ArgumentException">The <paramref name="destination"/>'s length is smaller than that of the <paramref name="source"/>.</exception>
+        [CLSCompliant(false)]
+        public static void ReverseEndianness(ReadOnlySpan<UInt128> source, Span<UInt128> destination) =>
+            ReverseEndianness(MemoryMarshal.Cast<UInt128, Int128>(source), MemoryMarshal.Cast<UInt128, Int128>(destination));
+
+        /// <summary>Copies every primitive value from <paramref name="source"/> to <paramref name="destination"/>, reversing each primitive by performing an endianness swap as part of writing each.</summary>
+        /// <param name="source">The source span to copy.</param>
+        /// <param name="destination">The destination to which the source elements should be copied.</param>
+        /// <remarks>The source and destination spans may overlap. The same span may be passed as both the source and the destination in order to reverse each element's endianness in place.</remarks>
+        /// <exception cref="ArgumentException">The <paramref name="destination"/>'s length is smaller than that of the <paramref name="source"/>.</exception>
+        public static void ReverseEndianness(ReadOnlySpan<Int128> source, Span<Int128> destination)
+        {
+            if (destination.Length < source.Length)
+            {
+                ThrowDestinationTooSmall();
+            }
+
+            if (Unsafe.AreSame(ref MemoryMarshal.GetReference(source), ref MemoryMarshal.GetReference(destination)) ||
+                !source.Overlaps(destination, out int elementOffset) ||
+                elementOffset < 0)
+            {
+                // Iterate from beginning to end
+                for (int i = 0; i < source.Length; i++)
+                {
+                    destination[i] = ReverseEndianness(source[i]);
+                }
+            }
+            else
+            {
+                // Iterate from end to beginning
+                for (int i = source.Length - 1; i >= 0; i--)
+                {
+                    destination[i] = ReverseEndianness(source[i]);
+                }
+            }
```

Comment:
same question as for the vectorized code path: is there a noticeable difference in iterating from the end vs from the start?
```suggestion
            // Iterate from end to beginning
            for (int i = source.Length - 1; i >= 0; i--)
            {
                destination[i] = ReverseEndianness(source[i]);
            }
```